k-distribution/src/main/scripts/bin/kx

```diff
@@ -0,0 +1,438 @@
+      ;;
+  
+      -p*=*)
+      keyval=${1#-p}
+      key=${keyval%%=*}
+      val=${keyval#*=}
+      var_name="parser_$key"
+      printf -v "$var_name" %s "$val"
+      ;;
+  
+      -c*=*)
+      keyval=${1#-c}
+      key=${keyval%%=*}
+      val=${keyval#*=}
+      var_name="config_var_$key"
+      params+=("$key")
+      tempFile="$(mktemp tmp."$key".pretty.XXXXXXXXXX)"
+      tempFiles+=("$tempFile")
+      printf %s "$val" > "$tempFile"
+      printf -v "$var_name" %s "$tempFile"
+      ;;
+  
+      -d|--directory)
+      dir="$2"
+      shift
+      ;;
+  
+      --dry-run)
+      cmdprefix="echo "
+      tempFiles=(${tempFiles[@]/$input_file})
+      outputMode=none
+      ;;
+  
+      --haskell-backend-command)
+      cmd="$2"
+      shift
+      ;;
+  
+      -h|--help)
+      print_usage
+      exit 0
+      ;;
+  
+      --io)
+      case "$2" in
+        on)
+        io=true
+        ;;
+  
+        off)
+        io=false
+        ;;
+  
+        *)
+        error 'Invalid value for --io. Should be "on" or "off".'
+        ;;
+      esac
+      shift
+      ;;
+  
+      --no-substitution-filtering)
+      filterSubst=false
+      ;;
+  
+      -o|--output)
+      case "$2" in
+        pretty) ;;
+        program) ;;
+        kast) ;;
+        binary) ;;
+        json) ;;
+        latex) ;;
+        kore) ;;
+        none) ;;
+        *)
+          error 'Invalid value for --output. Should be one of "pretty", "program", "kast", "binary", "json", "latex", "kore", or "none".'
+  	;;
+      esac
+      outputMode="$2"
+      shift
+      ;;
+  
+      --output-file)
+      outputFile="$2"
+      shift
+      ;;
+  
+      --parser)
+      var_name="parser_PGM"
+      printf -v "$var_name" %s "$2"
+      shift
+      ;;
+  
+      --pattern)
+      pattern="$2"
+      shift
+      ;;
+  
+      --search)
+      search=true
+      searchType='!'
+      ;;
+  
+      --search-all)
+      search=true
+      searchType='*'
+      ;;
+  
+      --search-final)
+      search=true
+      searchType='!'
+      ;;
+  
+      --search-one-or-more-steps)
+      search=true
+      searchType='+'
+      ;;
+  
+      --search-one-step)
+      search=true
+      searchType='1'
+      ;;
+  
+      --term)
+      term=true
+      ;;    
+  
+      -v|--verbose)
+      verbose=true
+      flags="$flags -v"
+      ;;
+  
+      -save-temps)
+      trap - INT TERM EXIT
+      flags="$flags -save-temps"
+      ;;
+  
+      --version)
+      kompile --version
+      exit 0
+      ;;
+  
+      --debugger)
+      cmdprefix="gdb --args "
+      ;;
+  
+      --statistics)
+      flags="$flags --statistics"
+      ;;
+  
+      --)
+      literal=true
+      ;;
+  
+      -*)
+      print_usage
+      exit 1
+      ;;
+  
+      *)
+      ARGV+=("$1")
+      ;;
+    esac
+  fi
+  shift
+done
+
+if $search; then
+  io=false
+fi
+
+if [[ "${#ARGV[@]}" -gt 1 ]]; then
+  error 'Cannot pass more than one configuration variable as a positional argument.'
+fi
```

Comment:
Just a styling argument, any single-line `if CONDITION; then COMMAND; fi` can be rewritten as `! CONDITION || COMMAND`. Up to you.